cardano-cli/src/Cardano/CLI/Shelley/Run/Pretty.hs

```diff
@@ -1,107 +1,180 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 
-module Cardano.CLI.Shelley.Run.Pretty (prettyTx) where
+-- | User-friendly pretty-printing for textual user interfaces (TUI)
+module Cardano.CLI.Shelley.Run.Pretty (friendlyTxBodyBS) where
 
-import           Cardano.Api
-                   (ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley))
+import           Cardano.Prelude hiding (undefined)
+import           Prelude (error)
+
+import           Cardano.Api as Api (AddressInEra (..),
+                   AddressTypeInEra (ByronAddressInAnyEra, ShelleyAddressInEra),
+                   ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley),
+                   ShelleyEra, TxBody, serialiseToBech32)
 import           Cardano.Api.Byron (TxBody (ByronTxBody))
-import           Cardano.Api.Shelley (TxBody (ShelleyTxBody))
+import           Cardano.Api.Shelley (TxBody (ShelleyTxBody), fromShelleyAddr)
+import           Cardano.Binary (Annotated)
 import           Cardano.CLI.Helpers (textShow)
+import qualified Cardano.Chain.UTxO as Byron
+import           Cardano.Ledger.Shelley as Ledger (ShelleyEra)
+import           Cardano.Ledger.ShelleyMA (MaryOrAllegra (Allegra, Mary), ShelleyMAEra)
 import qualified Cardano.Ledger.ShelleyMA.TxBody as ShelleyMA
-import           Cardano.Prelude
-import           Data.Aeson as JSON (Value, object, (.=))
-import           Data.Aeson.Encode.Pretty (Config (confCompare), defConfig, encodePretty')
+import           Data.Aeson as JSON (Object, Value (..), object, toJSON, (.=))
```

Comment:
When qualifying and import, be sure to search the repo to see how its usually qualified. In this instance we usually qualify `Data.Aeson` as `Aeson`